src/lib/comms/sol-lwm2m.c

```diff
@@ -4830,6 +5150,14 @@ sol_lwm2m_client_start(struct sol_lwm2m_client *client)
 
     return 0;
 
+err_access_control:
+    if (client->supports_access_control) {
+        ctx = find_object_ctx_by_id(client, ACCESS_CONTROL_OBJECT_ID);
+        while ((instance = sol_vector_get(&ctx->instances, 0)) != NULL) {
```

Comment:
You can always do a reverse loop and delete, using `SOL_VECTOR_FOREACH_REVERSE_IDX`.